Error installing with `cargo install cargo-binstall`

Hi! Just tried installing `cargo-binstall` with `cargo` and I am getting an error:
```sh
> cargo install cargo-binstall
> ....
error[E0119]: conflicting implementations of trait `cargo_toml::AbstractFilesystem` for type `&drivers::crates_io::vfs::Vfs`
  --> /home/esp/.cargo/registry/src/github.com-1ecc6299db9ec823/binstalk-0.2.0/src/drivers/crates_io/vfs.rs:45:1
   |
45 | impl AbstractFilesystem for &Vfs {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: conflicting implementation in crate `cargo_toml`:
           - impl<T> AbstractFilesystem for &T
             where T: AbstractFilesystem, T: ?Sized;

For more information about this error, try `rustc --explain E0119`.
error: could not compile `binstalk` due to previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `cargo-binstall v0.13.1`, intermediate artifacts can be found at `/tmp/cargo-installNXWXW9`
```
I've tried installing it with stable (rustc 1.63.0 (4b91a6ea7 2022-08-08)) and nightly (rustc 1.66.0-nightly (e7119a030 2022-09-22))

--- a/crates/binstalk/Cargo.toml
+++ b/crates/binstalk/Cargo.toml
@@ -13,7 +13,7 @@ license = "GPL-3.0"
 async-trait = "0.1.57"
 bytes = "1.2.1"
 bzip2 = "0.4.3"
-cargo_toml = "0.11.5"
+cargo_toml = "0.12.0"
 clap = { version = "3.2.22", features = ["derive"] }
 compact_str = { version = "0.6.0", features = ["serde"] }
 crates_io_api = { version = "0.8.1", default-features = false }

--- a/crates/binstalk/src/drivers/crates_io/vfs.rs
+++ b/crates/binstalk/src/drivers/crates_io/vfs.rs
@@ -41,9 +41,3 @@ impl AbstractFilesystem for Vfs {
         Ok(self.0.get(&*rel_path).map(Clone::clone).unwrap_or_default())
     }
 }
-
-impl AbstractFilesystem for &Vfs {
-    fn file_names_in(&self, rel_path: &str) -> io::Result<HashSet<Box<str>>> {
-        (*self).file_names_in(rel_path)
-    }
-}

--- a/crates/binstalk/tests/parse-meta.rs
+++ b/crates/binstalk/tests/parse-meta.rs
@@ -24,7 +24,7 @@ fn parse_meta() {
         &[Product {
             name: Some("cargo-binstall".to_string()),
             path: Some("src/main.rs".to_string()),
-            edition: Some(cargo_toml::Edition::E2021),
+            edition: cargo_toml::Edition::E2021,
             ..Default::default()
         },],
     );

